src/classes/QueryBuilder.cls

```diff
@@ -70,107 +83,179 @@ public without sharing class QueryBuilder extends NebulaCore implements IQueryBu
+
+        this.query += String.join(queryStringSpecifiers, ',');
+    }
+
+    private String getFieldNameByRelationship(SObjectField field) {
+        SObjectType fieldDerivesFrom = DescribeUtils.getSObjectType(field);
+        Boolean isSameObjectField = fieldDerivesFrom == this.sobjectType;
+        if(isSameObjectField) {
+            return field.getDescribe().getName();
+        } else {
+            Boolean derivesFromCustomObject = fieldDerivesFrom.getDescribe().isCustom();
+            String searchField = field.getDescribe().getName();
+            for(String fieldName : this.sobjectTypeFieldMap.keySet()) {
+                String withoutCustomDelimiter = fieldName;
+                if(fieldName.right(3) == '__c' && !derivesFromCustomObject) {
+                    withoutCustomDelimiter = fieldName.left(fieldName.length() - 3);
+                }
+                if(fieldDerivesFrom.getDescribe().getName() == withoutCustomDelimiter) {
+                    return this.sobjectTypeFieldMap.get(fieldName).getDescribe().getRelationshipName() + '.' + searchField;
+                }
+            }
+            return searchField;
+        }
+    }
+
+    @testVisible
+    private String getQuery() {
+        this.query = 'SELECT ' + this.getQueryFieldString()
+            + this.getFromClauseString()
             + this.getWhereClauseString()
             + this.getOrderByString()
             + this.getLimitCountString()
             + this.getForUpdateString();
 
-        Logger.addEntry(this, query);
+        return this.query;
+    }
 
-        return query;
+    private String getFromClauseString() {
+        return ' FROM ' + this.sobjectType;
     }
 
-    private String getSearchQueryString(String searchTerm, QuerySearchGroup searchGroup) {
-        String query = 'FIND ' + new QueryArgumentFormatter(searchTerm).getValue()
-            + '\nIN ' + searchGroup.name().replace('_', ' ')
-            + '\nRETURNING ' + this.sobjectType + '('
+    protected virtual String getSearchQuery(String searchTerm, QuerySearchGroup searchGroup) {
+        this.query = 'FIND ' + new QueryArgumentFormatter(searchTerm.toLowerCase()).getValue()
+            + ' IN ' + searchGroup.name().replace('_', ' ')
+            + ' RETURNING ' + this.sobjectType + '('
                 + this.getQueryFieldString()
                 + this.getWhereClauseString()
                 + this.getOrderByString()
                 + this.getLimitCountString()
             + ')';
 
-        if(this.forUpdate) Logger.addEntry(this, 'SOSL Search Query method flagged as FOR UPDATE. SOSL cannot use FOR UPDATE, ignoring');
-        if(this.filterScope != null) Logger.addEntry(this, 'SOSL Search Query method flagged as USING SCOPE ' + this.filterScope + '. SOSL cannot use USING SCOPE, ignoring');
-
-        Logger.addEntry(this, query);
+        if(this.forUpdate) System.debug(LoggingLevel.WARN, 'SOSL Search Query method flagged as FOR UPDATE. SOSL cannot use FOR UPDATE, ignoring');
```

Comment:
Ah again - this is just my bad.  I missed that once in a selective staging many commits ago and haven't backed it out.  Because we're not using the whole framework (mostly just the SObjectRepo and associated classes) there are some inconsistencies in the files - inconsistencies that have affected you in readability, so I am very sorry about that!